cmake/support_pllgpu.cmake

```diff
@@ -0,0 +1,94 @@
+###############################################################################################
+# This file provides macros to process CUDA, CUDAToolkit, cuQuantum and MPI external libraries.
+###############################################################################################
+
+# Include this only once
+include_guard()
+
+set(CRAY_MPICH OFF)
```

Comment:
Is this really needed here? I still think we need to find a way to avoid keeping system-specific options and flags in here.

Reply:
yea, we can remove it and updated in the MPI backend PR. It's for the GTL linkage. We can just check if there is GTL lib. If yes, we link against it.